client/my-sites/sites/sites.jsx

```diff
@@ -1,62 +1,77 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component, PropTypes } from 'react';
 import page from 'page';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import Card from 'components/card';
 import Main from 'components/main';
-import observe from 'lib/mixins/data-observe';
 import SiteSelector from 'components/site-selector';
 import { addSiteFragment } from 'lib/route';
+import { getSelectedSite } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import isPrivateSite from 'state/selectors/is-private-site';
+import isSiteUpgradeable from 'state/selectors/is-site-upgradeable';
 
-export default React.createClass( {
-	displayName: 'Sites',
+class Sites extends Component {
+	static propTypes = {
+		getSiteSelectionHeaderText: PropTypes.func,
```

Comment:
I've noticed some expected props are missing from the `propTypes`, for example `sourcePath`. Also, in your other PRs I noticed you were adding `translate` too, but it's missing here?